sql/mysqld.cc

```diff
@@ -4082,6 +4082,9 @@ static int init_common_variables()
     /* MyISAM requires two file handles per table. */
     wanted_files= (10 + max_connections + extra_max_connections +
                    tc_size * 2);
+#if defined(HAVE_POOL_OF_THREADS) && !defined(__WIN__)
+    wanted_files+= threadpool_size;
```

Comment:
Wlad also requested to do it under `if (thread_handling == 2)`. FWICT thread_handling can't change at run-time and we don't allocate anything otherwise.

Reply:
weirdly, there is no constant for thread_handling == SCHEDULER_POOL_OF_THREADS . This is probably what we inherited from MySQL, where Enterprise Threadpool scheduler would be a dynamically loaded scheduler. Maybe we should introduce that constant in scheduler.h, since we do not load anything dynamically.

Reply:
but as it stands today 

#if defined(HAVE_POOL_OF_THREADS) && !defined(_WIN32)
  // add epoll or keven fd for each threadpool group, in case pool of threads is used
  wanted_file += (thread_handling > SCHEDULER_NO_THREADS)? 0 : threadpool_size;
#endif

should be sufficient

Reply:
Note  - test suite fixup was required, because it did not run cleanly on buildbot ,e.g 
http://buildbot.askmonty.org/buildbot/builders/kvm-bintar-centos5-amd64/builds/10350/steps/test/logs/stdio

The fixup
https://github.com/MariaDB/server/commit/23804dd1dd0585b91ab313aa3b7dc3a4b1e272e9

Reply:
thanks. Updated with the fixup, thread_handling condition and style fixes